integration-tests/tests/src/tests/mixed.ts

```diff
@@ -220,23 +345,882 @@ describe("Mixed", () => {
       expect(value.nullable_list[4]).equals(5, "Should be equal 5");
     });
   });
-  describe("Mixed arrays", () => {
-    openRealmBefore({ schema: [MixedSchema] });
-    it("throws when creating an array of multiple values", function (this: RealmContext) {
-      const objectsBefore = this.realm.objects(MixedSchema.name);
-      expect(objectsBefore.length).equals(0);
-
-      // check if the understandable error message is thrown
-      expect(() => {
+
+  describe("Collection types", () => {
+    openRealmBeforeEach({
+      schema: [
+        MixedSchema,
+        MixedAndEmbeddedSchema,
+        MixedWithDefaultCollectionsSchema,
+        CollectionsOfMixedSchema,
+        EmbeddedObjectSchema,
+        TestObject,
+      ],
+    });
+
+    function expectMatchingFlatList(value: unknown) {
+      expect(value).instanceOf(Realm.List);
+      const list = value as Realm.List<any>;
```

Comment:
It's frustrating that `chai`'s return type doesn't `assert` here 😕 We should probably look into using a more TypeScript friendly assert library for our tests ...

Reply:
Yeah I know, a bit strange that they don't do that. I've extracted it to a custom helper that wraps the `instanceOf` as you suggested 👍